src/mycollections/DynamicArray.java

```diff
@@ -1,59 +1,125 @@
 package mycollections;
 
-public class DynamicArray<E>  implements MyList<E> {
-
-	@Override
-	public int size() {
-		return 0;
-	}
-
-	@Override
-	public boolean isEmpty() {
-		return false;
-	}
-
-	@Override
-	public boolean contains(E element) {
-		return false;
-	}
-
-	@Override
-	public int indexOf(E element) {
-		return 0;
-	}
-
-	@Override
-	public int lastIndexOf(E element) {
-		return 0;
-	}
-
-	@Override
-	public E get(int index) {
-		return null;
-	}
-
-	@Override
-	public E set(int index, E element) {
-		return null;
-	}
-
-	@Override
-	public boolean add(E e) {
-		return false;
-	}
-
-	@Override
-	public void add(int index, E element) {
-
-	}
-
-	@Override
-	public E remove(int index) {
-		return null;
-	}
-
-	@Override
-	public boolean remove(E element) {
-		return false;
-	}
+public class DynamicArray<E> implements MyList<E> {
+
+    private E[] array;
+    private final int DEFAULT_SIZE = 16;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public DynamicArray() {
+        array = (E[]) new Object[DEFAULT_SIZE];
+    }
+
+    @SuppressWarnings("unchecked")
+    public DynamicArray(int initCapacity) {
+        array = (E[]) new Object[initCapacity];
+
+    }
+
+    @Override
+    public int size() {
+        return this.size;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return this.size == 0;
+    }
+
+    @Override
+    public boolean contains(E element) {
```

Comment:
You can use the indexOf() method to check if the element exists in the list or not.